File: flutter_render/lib/src/rowscol/Row.dart
import 'package:neneui_render/src/enum.dart';
import 'package:neneui_render/src/parser/Core.dart';
import 'package:shadcn_flutter/shadcn_flutter_experimental.dart';

class dRow {
  static Widget run({
    required BuildContext context,
    required Map<String, dynamic> data,
    required Function reRender,
    required Function reRenderList,
    required Function event,
  }) {
    if (data['name'] == "Row") {
      event(Events.REGISTER_ID, {'id': data['id'], 'props': data['props']});

      final rawChildren = data['props']['children'];

      final childrenList = rawChildren is List ? rawChildren : [rawChildren];

      final children = <Widget>[];

      for (var ui in childrenList) {
        if (data['props']['foreach']) {
          children.addAll(reRenderList(ui));
        } else {
          children.add(reRender(ui));
        }
      }

      return Row(
        mainAxisAlignment: CoreParser.parseW(data['props']['mainAxis']),
        crossAxisAlignment: CoreParser.crossParse(data['props']['crossAxis']),
        children: children,
      );
    } else {
      return SizedBox();
    }
  }
}

File: lib/childrens/Row.ts
import { CrossAxis, Empty, MainAxis } from "../widgets";

export interface RowProps {
    mainAxisAlignment?: MainAxis,
    crossAxisAlignment?: CrossAxis,
    foreach?: boolean,
    children: any[]
}

export function Row(id: string, {
    mainAxisAlignment = MainAxis.start,
    crossAxisAlignment = CrossAxis.start,
    foreach = false,
    children = [Empty()]
}: RowProps) {
    return {
        id,
        name: "Row",
        props: {
            mainAxis: mainAxisAlignment,
            crossAxis: crossAxisAlignment,
            foreach,
            children: children
        }
    }
}